### PR TITLE
Minor linting updates

### PR DESCRIPTION
**Pull request by lemonmade**
This PR makes a few small JavaScript linting/ styleguide updates:

- Made it so that `require-flow` allows `@flow` to appear in a line comment (which is apparently allowed now).
- Removed the requirement to omit the braces for arrow functions that have only a return statement and updated the guidance for them to suggest (but not require) omitting the braces for simple return-only arrow functions.
- Turned off linting rules that enforced sorting of properties in React components. This was annoying in my experience and doesn't have the granularity to make it work well with class properties.
- Turned off a rule that required you to use a default export when your module exports only one identifier. IMO some modules make more sense as groups of named exports (usually utility-like things), even if you currently only have one export from that group.

Thoughts? @GoodForOneFare @bouk @tessalt @Shopify/fed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1353,27 +1351,13 @@ npm run lint
...
   // good
-  [1, 2, 3]
-    .map((x) => (x * x) + 1)
-    .filter((x) => x < 6)
-    .filter((a) => (
-      a &&
-      really &&
-      (long || expression)
-    ));
+  [1, 2, 3].map((x) => (x * x) + 1).filter((x) => x < 6);
```
Comment (GoodForOneFare) [inline README.md]: Should `.filter` be on a new line?

File: packages/eslint-plugin-shopify/lib/config/rules/ecmascript-6.js [hunk 2]
```diff
@@ -2,19 +2,19 @@
 
 module.exports = {
   // Require braces in arrow function body
-  'arrow-body-style': ['error', 'as-needed'],
+  'arrow-body-style': 'off',
```
Comment (tessalt) [inline packages/eslint-plugin-shopify/lib/config/rules/ecmascript-6.js]: 😂

Comment (tessalt) [inline packages/eslint-plugin-shopify/lib/config/rules/ecmascript-6.js]: I actually meant that as tears of joy, not laughter.

**Comment by GoodForOneFare** [comment]
👍 LGTM.

File: packages/eslint-plugin-shopify/lib/config/rules/react.js [hunk 3]
```diff
@@ -76,19 +76,7 @@ module.exports = {
...
-        'static-methods',
-        'statics',
-        'constructor',
-        'lifecycle',
-        'everything-else',
-        'render',
-      ],
-    },
-  ],
+  'react/sort-comp': 'off',
```
Comment (bouk) [inline packages/eslint-plugin-shopify/lib/config/rules/react.js]: :burn: this rule

File: packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js [hunk 4]
```diff
@@ -50,7 +50,7 @@ module.exports = {
   // Specify the maximum number of statements allowed per line
   'max-statements-per-line': ['warn', {max: 2}],
   // Require a capital letter for constructors
-  'new-cap': 'off',
+  'new-cap': ['error', {newIsCap: true, capIsNew: false}],
```
Comment (bouk) [inline packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js]: why did you change this but then still disable is in `esnext.js`?

Comment (lemonmade) [inline packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js]: Before now, if you were using the `es5` config, you wouldn't get any version of this rule at all (since the babel plugin is only brought in for `esnext`).

**Comment by tessalt** [comment]
can we get this released?

**Comment by lemonmade** [comment]
@tessalt 13.0.0 released with all the linting changes, let me know if you have nay issues!
